### PR TITLE
RST backtick refactor manual.rst

### PR DESCRIPTION
**Pull request by quantimnot**
Added `.. default-role:: code` directive.

Replaced double backticks with single backticks and realign shifted table columns.

### CODE REVIEW

**Comment by timotheecour** [comment]
* followup https://github.com/nim-lang/Nim/pull/17028
* see https://github.com/nim-lang/Nim/issues/17260; how about we wait a bit until we know whether https://github.com/nim-lang/Nim/issues/17260 can be fixed quickly enough so that this PR doesn't need patching
* to view generated html fast, use: `nim rst2html -r --lib:lib --doccmd:skip doc/manual.rst`

**Comment by timotheecour** [comment]
@quantimnot how about, for this PR as well as https://github.com/nim-lang/Nim/pull/17258, keep the double backticks when the content contains a `\` somewhere inside.

Then we can move forward with this PR without waiting on https://github.com/nim-lang/Nim/pull/17315, and if/when https://github.com/nim-lang/Nim/pull/17315 is merged, we can revisit those small number of cases in subsequent PR.

File: doc/manual.rst [hunk 1]
```diff
@@ -360,7 +362,7 @@ contain the following `escape sequences`:idx:\ :
   ``\\``                   `backslash`:idx:
   ``\"``                   `quotation mark`:idx:
   ``\'``                   `apostrophe`:idx:
-  ``\`` '0'..'9'+          `character with decimal value d`:idx:;
+  ``\\`` '0'..'9'+         `character with decimal value d`:idx:;
```
Comment (timotheecour) [inline doc/manual.rst:365]:
```suggestion
  ``\`` '0'..'9'+          `character with decimal value d`:idx:;
```

File: doc/manual.rst [hunk 2]
```diff
@@ -471,7 +472,7 @@ literals:
...
```
Comment (timotheecour) [inline doc/manual.rst:475]:
```suggestion
  ``\`` '0'..'9'+          `character with decimal value d`:idx:;
```

File: doc/manual.rst [hunk 3]
```diff
@@ -6073,21 +6074,20 @@ avoid ambiguity when there are multiple modules with the same path.
...
-   scope of this document, its semantics are: *Use the search path to look for
-   module name but ignore the standard library locations*.
-   In other words, it is the opposite of ``std``.
+1. `std`: The `std` pseudo directory is the abstract location of Nim's standard
+library. For example, the syntax `import std / strutils` is used to unambiguously
+refer to the standard library's `strutils` module.
+2. `pkg`: The `pkg` pseudo directory is used to unambiguously refer to a Nimble
+package. However, for technical details that lie outside the scope of this document,
+its semantics are: *Use the search path to look for module name but ignore the standard
+library locations*. In other words, it is the opposite of `std`.
```
Comment (timotheecour) [inline doc/manual.rst:6083]:
```suggestion
1. `std`: The `std` pseudo directory is the abstract location of Nim's standard
   library. For example, the syntax `import std / strutils` is used to unambiguously
   refer to the standard library's `strutils` module.
2. `pkg`: The `pkg` pseudo directory is used to unambiguously refer to a Nimble
   package. However, for technical details that lie outside the scope of this document,
   its semantics are: *Use the search path to look for module name but ignore the standard
   library locations*.
   In other words, it is the opposite of `std`.
```

File: doc/manual.rst [hunk 4]
```diff
@@ -7690,38 +7690,38 @@ string expressions in general:
...
 
 Threads
 =======
 
-To enable thread support the ``--threads:on`` command-line switch needs to
-be used. The ``system`` module then contains several threading primitives.
-See the `threads <threads.html>`_ and `channels <channels_builtin.html>`_ modules
+To enable thread support the `--threads:on` command-line switch needs to
+be used. The `system` module then contains several threading primitives.
+See the `threads <threads.html>`_ and `channels <channels.html>`_ modules
```
Comment (timotheecour) [inline doc/manual.rst:7724]:
```suggestion
See the `threads <threads.html>`_ and `channels <channels_builtin.html>`_ modules
```

**Review by timotheecour — APPROVED** [review]
LGTM; I've checked thoroughly in 2 different ways:
* checking the diff
* checking the rendered html (nim rst2html) and going page by page carefully, comparing the html before PR and after PR; the pages align and any difference would be obvious to the eye; only required 1 place to offset manually (explained below)
* this PR was rebased but incorrectly dropped the (tiny) diff from 2 recent PR's : #17330 and #17257 ; this can be fixed up in a followup PR
* also noted `\\` vs `\` (ie a \ was inserted wrongly) which can be fixed later
* everything else was identical, except for things like:
```nim
## ``foo``
=>
## `foo`
```
in code blocks, which render differently (since its verbatim); but this is an expected, desirable change.

**Comment by timotheecour** [comment]
I've merged this after a careful review (see previous comment) to avoid this PR from bit-rotting and needing to be rebased again and again; I've noted the followups needed.

**Comment by timotheecour** [comment]
@quantimnot thanks for doing this! could you please followup with a PR to address remaining points? (can be bundled with the fixups needed for https://github.com/nim-lang/Nim/pull/17258#issuecomment-802526085)
